### PR TITLE
refactor(testing): moved mock client out of source

### PR DESCRIPTION
**Pull request by sg-s**
## changes

fixes: https://github.com/formiclabs/deeporigin-client/pull/33#discussion_r1619622890

### CODE REVIEW

File: src/managed_data/client.py
Comment (sg-s) [inline src/managed_data/client.py:1]: moved MockClient out of src code because this isn't needed in operation of code

File: tests/mock_client.py
Comment (sg-s) [inline tests/mock_client.py:1]: created a new file to contain the mock client

**Review by sg-s — COMMENTED** [review]
self-review

**Review by jonrkarr — APPROVED** [review]
Looks great

**Pull request merged**